modules/nlu/src/backend/train-repo-router.ts

```diff
@@ -74,14 +74,17 @@ const createRepositoryRouter = (app: NLUApplication): Router => {
     if (error) {
       return res.status(400).json({ error: `Training status body is invalid: ${error.message}` })
     }
-
-    await repo.set({ botId, language: lang }, value)
+    await repo.inTransaction(trx => {
```

Comment:
those are the only cases where I wouldnt mind updating the training repo while not in a transaction (In case we want to un-jam a deadlock or something)

But this is fine

Reply:
yeah some of the transactions are unnecessary but if we want to force set/deletes to go through transactions that's the price to pay